docs/guides/validation.md

```diff
@@ -205,8 +267,52 @@ This will debounce every async call with a 500ms delay. You can even override th
 
 > This will run `onChangeAsync` every 1500ms while `onBlurAsync` will run every 500ms.
 
+## Validating fields that depend on each other
 
-## Adapter-Based Validation
+Sometimes when validating a field value, we need to access the value of another field. For example a "repeat password" field that needs to be identical to a "password" field. On top of receiving the field value, each validation callback receives the `fieldAPI` object which has access to the `formAPI` object, which contains all the fields values.
+
+Here is an example
+```tsx
+<form.Field
+  name="password"
+  children={(field) => (
+    <>
+      <label htmlFor={field.name}>Password:</label>
+      <input
+        name={field.name}
+        value={field.state.value}
+        type="password"
+        onChange={(e) => field.handleChange(e.target.value)}
+      />
+      <FieldInfo field={field} />
+    </>
+  )}
+/>
+<form.Field
+  name="repeat_password"
+  onChange={(value, fieldAPI) => {
+    if (value !== fieldAPI.form.store.state.values.password) {
+      return "Passwords do not match";
+    }
+
+  }}
+  children={(field) => (
+    <>
+      <label htmlFor={field.name}>Repeat password:</label>
+      <input
+        name={field.name}
+        value={field.state.value}
+        type="password"
+        onChange={(e) => field.handleChange(e.target.value)}
+      />
+      <FieldInfo field={field} />
+    </>
+  )}
+/>
+```
+
+
```

Comment:
I'd like to remove this temporarily and move this into an issue/discussion.

Right now, I'm not convinced this is a reliable mechanism for validating across fields. **If** this works, I'm almost certain it's due to a performance problem that we'll aim to fix using something akin to this:

https://houseform.dev/guides/performance-optimizations.html

Reply:
Makes sense! I've removed this part and created [a discussion](https://github.com/TanStack/form/discussions/494).